lint/rule_template_required_variables.go

```diff
@@ -0,0 +1,43 @@
+package lint
+
+import (
+	"strings"
+)
+
+type TemplateRequiredVariablesRuleSettings struct {
+	Variables []string `yaml:"variables"`
+}
+
+func NewTemplateRequiredVariablesRule(config *TemplateRequiredVariablesRuleSettings, requiredMatchers *TargetRequiredMatchersRuleSettings) *DashboardRuleFunc {
+	return &DashboardRuleFunc{
+		name:        "template-required-variables-rule",
+		description: "Checks that the dashboard has a template variable for required variables or matchers that use variables",
+		stability:   "experimental",
+		fn: func(d Dashboard) DashboardRuleResults {
+			r := DashboardRuleResults{}
+
+			template := getTemplateDatasource(d)
+			if template == nil || template.Query != Prometheus {
+				return r
+			}
+
+			// Convert the config.variables to a map to leverage uniqueness...
+			variables := make(map[string]bool)
```

Comment:
Same concern over missing config, resulting in a nil here. Probably worth making a test to ensure that case is tested.